Fill.py

```diff
@@ -23,7 +23,8 @@ def sweep_from_pool(base_state: CollectionState, itempool: typing.Sequence[Item]
 
 
 def fill_restrictive(world: MultiWorld, base_state: CollectionState, locations: typing.List[Location],
-                     itempool: typing.List[Item], single_player_placement: bool = False, lock: bool = False) -> None:
+                     itempool: typing.List[Item], single_player_placement: bool = False, lock: bool = False,
+                     swap: bool = True, on_place: typing.Callable[[Location], None] = None) -> None:
```

Comment:
mypy does not complain, but shouldn't that be Optional[Callable] if it defaults to None?
```suggestion
                     swap: bool = True, on_place: typing.Optional[typing.Callable[[Location], None]] = None) -> None:
```

Reply:
Prior to 3.10 docs it was written that this Optional is optional, but current python docs state that the optional Optional is now desired for explicitness sake, so yes. I guess mypy still allows 3.9 and earlier conformity